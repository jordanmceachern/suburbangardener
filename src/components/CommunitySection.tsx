import React from "react";
import Link from "next/link";

interface CommunitySectionProps {
  className?: string;
}

export default function CommunitySection({
  className = "",
}: CommunitySectionProps) {
  return (
    <section
      className={`p-6 rounded-lg bg-primary-100 dark:bg-secondary-900 border border-primary-200 dark:border-secondary-800 ${className}`}
    >
      <div className="text-center">
        <h2 className="text-xl font-bold mb-3 text-neutral-900 dark:text-neutral-100">
          Join Our Gardening Community
        </h2>
        <p className="text-primary-700 dark:text-neutral-300 mb-6">
          Sign up for exclusive access to premium gardening tools, personalized
          plant care guides, and more!
        </p>
        <div className="flex flex-col sm:flex-row gap-4 justify-center">
          <Link
            href="/register"
            className="bg-primary-600 hover:bg-primary-700 text-white font-medium py-3 px-6 rounded-lg transition-colors"
          >
            Get Started - Sign Up
          </Link>
          <Link
            href="/login"
            className="bg-white dark:bg-tertiary-800 hover:bg-neutral-50 dark:hover:bg-tertiary-900 text-neutral-600 dark:text-neutral-100 font-medium py-3 px-6 rounded-lg border border-primary-300 dark:border-none transition-colors"
          >
            Already a Member? Log In
          </Link>
        </div>
      </div>
    </section>
  );
}
