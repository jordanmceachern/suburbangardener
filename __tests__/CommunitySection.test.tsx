import { render } from "@testing-library/react";
import CommunitySection from "@/components/CommunitySection";

describe("CommunitySection Component", () => {
  it("should match DOM snapshot", () => {
    const { container } = render(<CommunitySection />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should maintain CSS classes for community section layout", () => {
    const { container } = render(<CommunitySection />);

    // Check for section-related classes
    const hasSectionClasses =
      container.innerHTML.includes("section") ||
      container.innerHTML.includes("community") ||
      container.innerHTML.includes("text-center") ||
      container.innerHTML.includes("flex") ||
      container.innerHTML.includes("grid");

    expect(hasSectionClasses).toBe(true);
  });
});
