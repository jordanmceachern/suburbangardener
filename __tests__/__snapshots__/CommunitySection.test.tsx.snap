// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`CommunitySection Component should match DOM snapshot 1`] = `
<section
  class="p-6 rounded-lg bg-primary-100 dark:bg-slate-600 border border-primary-200 dark:border-slate-500 "
>
  <div
    class="text-center"
  >
    <h2
      class="text-xl font-bold mb-3 text-neutral-900 dark:text-neutral-100"
    >
      Join Our Gardening Community
    </h2>
    <p
      class="text-start sm:text-center text-primary-700 dark:text-neutral-300 mb-6"
    >
      Sign up for exclusive access to premium gardening tools, personalized plant care guides, and more!
    </p>
    <div
      class="flex flex-col sm:flex-row gap-4 justify-center"
    >
      <a
        class="bg-primary-600 hover:bg-primary-700 text-white font-medium py-3 px-6 rounded-lg transition-colors"
        href="/register"
      >
        Get Started - Sign Up
      </a>
      <a
        class="bg-white dark:bg-tertiary-800 hover:bg-neutral-50 dark:hover:bg-tertiary-900 text-neutral-600 dark:text-neutral-100 font-medium py-3 px-6 rounded-lg border border-primary-300 dark:border-none transition-colors"
        href="/login"
      >
        Already a Member? Log In
      </a>
    </div>
  </div>
</section>
`;
